simulator/src/main/java/com/hedera/block/simulator/BlockStreamSimulatorApp.java

```diff
@@ -155,8 +98,9 @@ public boolean isRunning() {
         return isRunning.get();
     }
 
-    /** Stops the block stream simulator. */
+    /** Stops the Block Stream Simulator and closes off all grpc channels. */
     public void stop() {
+        publishStreamGrpcClient.shutdown();
         isRunning.set(false);
```

Comment:
What happens if shutdown fails for some reason and does not close the channel, while the variable is already set on false? `isRunning` should indicate whether the simulator is fully stopped and that is only true after all steps (before `isRunning.set(false)`) have passed.

Reply:
If the shutdown fails, the channel would be in a broken state, if we can recover from that, then I would personally consider the app to be running, but if we cannot, then it technically is running yes, but in a broken state. Maybe this discussion is to be viewed in more detail. My concern is that we will start to shutdown the channel and logic that would poll if the app is running will be mislead by the boolean. Maybe a better approach would be to have states for the app rather than a simple boolean. States like, 'initializing', `running`, `paused`, `recovering` `stopping`, `stopped` etc. etc. in order for callers to know with more certainty what is the status and take appropriate measure. Of course all of this is not within our scope here, but I think it is a good and important topic to discuss for the future. I wanted just to discuss this briefly and bring attention.

Reply:
Yes, good point about the states. Will open a EPIC with improvements for the simulator based on the feedback ✅